### PR TITLE
Tiling rope and its geometry

### PR DESCRIPTION
**Pull request by scythianfuego**
##### Description of change
Hi, this adds an ability not to stretch the texture in the RopeGeometry. This allow to tile the rope easier (using PoT texture).
Could be useful to create any kind of roads/ropes.
Demo here - top to bottom: original texture, tiled rope, stretched rope (as now): http://128.199.61.220/tilingrope/

Also I've added TilingRope class (which is almost the same as SimpleRope, so this could be moved there instead)

Note: the reason there is a textureScale parameter is to provide a way for downsampling, so it is not related to rope width (thickness)!

~~Actually, it can be linked to rope width, provided we leave TilingRope class to set it from texture.
I'll update pull request to be consistent.~~ -> Not possible without RopeGeometry knowing actual texture width.

##### Pre-Merge Checklist

- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

**Comment by ivanpopelyshev** [comment]
Just notifying that team is a bit slow right now, but its fine.

My personal opinion: need an options for those settings, too many params in constructor.

@scythianfuego, you dont have to modify it, I'll discuss it with guys when they have time, your code is in good hands ;)

**Comment by scythianfuego** [comment]
No worries, I understand the team should focus on 5.0 release. I have no problems modifying anything here as I don't know what meshes and their implementation should look like as a a part of the big picture.

**Comment by bigtimebuddy** [comment]
@scythianfuego first off, great feature idea! Sorry that it's taken us so long to consider this. We've mostly been trying to smooth out all the rough edges with v5, which takes time unfortunately.

Overall, I'm not sure about the API approach.  Instead of creating a new class (`TilingRope`), I'd like to see `textureScale` (defaults to 1) in RopeGeometry and SimpleRope. Seems more like a feature to existing Rope, then a completely new class. Also, I don't think the `stretch` flag is necessary.

Is this something you'd want to take another swing at? Also, we can close the PR and I'm sure someone else might take up the effort.  Let me know how you'd like to move forward.

**Comment by scythianfuego** [comment]
> something you'd want to take another swing at
yes, no worries, i'll have a look at it tomorrow

File: packages/mesh-extras/src/geometry/RopeGeometry.js [hunk 1]
```diff
@@ -164,7 +196,8 @@ export default class RopeGeometry extends MeshGeometry
             }
 
             const perpLength = Math.sqrt((perpX * perpX) + (perpY * perpY));
-            const num = this.width / 2; // (20 + Math.abs(Math.sin((i + this.count) * 0.3) * 50) )* ratio;
+            const num = this.textureScale * this.width / 2;
+            // (20 + Math.abs(Math.sin((i + this.count) * 0.3) * 50) )* ratio;
```
[the remark this answers is not in the document]
Comment (scythianfuego) [reply packages/mesh-extras/src/geometry/RopeGeometry.js]: sure, cleaned up comments

**Comment by scythianfuego** [comment]
Updated.

Also, to add some demos, that's how i'm using this in my game to draw water surface top down: https://pixiplayground.com/#/edit/DJjJF2NsgdvRqe6Ev0lGC (adapted from #5266 )
Elastomania style grass could also be made like this.

**Comment by scythianfuego** [comment]
@ivanpopelyshev Could you have a look, please?

**Comment by bigtimebuddy** [comment]
Thanks @scythianfuego great PR and appreciate the flexibility. Sorry it took so long :)

**Pull request merged**